pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -124,20 +180,19 @@ func (cl *Client) Run(stop <-chan struct{}) {
 	t0 := time.Now()
 	scope.Info("Starting Pilot K8S CRD controller")
 
-	go func() {
-		if !cache.WaitForCacheSync(stop, cl.HasSynced) {
-			scope.Error("Failed to sync Pilot K8S CRD controller cache")
-			return
-		}
-		scope.Info("Pilot K8S CRD controller synced ", time.Since(t0))
-		cl.queue.Run(stop)
-	}()
-
+	if !cache.WaitForCacheSync(stop, cl.informerSynced) {
+		scope.Error("Failed to sync Pilot K8S CRD controller cache")
+		return
+	}
+	cl.SyncAll()
```

Comment:
Why is it better to skip updates then SyncAll here? I get for the k8s controller, but not or crd controller.. we still have the same amount of work to do?

Reply:
Without `SyncAll ` here, let's take an example here:

If we have total 100 configs here, without syncAll, let's say 100 configs enqueued and but only 50 is added to inboundedUpdate, and the other 50 has not been run. Then the time we mark istiod ready is not right, in this case, sidecar could get partial xds, which is dangerous.

Reply:
can we just enqueue some signal marks us as ready? then we know once we have processed that all 100 configs were processed as well. I think we do that in other places. That seems a bit simpler

Reply:
This will require the queue to be stateful.  And there is a pr replacing the queue with k8s rate limiting queue.  I am afraid this is not a good idea

Reply:
I am not sure I understand the issue. Regardless of the queue implementation, its always ordered. So when we dequeue a "ready" signal we can mark ourselves ready?

Reply:
I think i understand what you mean, but the tricky part is that will require a lock because SyncAll and other resource event handler can enqueue some tasks concurrently. That seems super hard to do , here the solution maybe do some little percentage of duplicate event handling, but from the sync view, it is the easiest.